### PR TITLE
Type definition via WFS-DescribeFeatureType

### PR DESCRIPTION
**Pull request by ahennr**
This PR implements `readData` of `geostyler-data` to fetch type definitions of a feature type via a `DescribeFeatureType` request (Currently, WFS-server needs to send CORS headers).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,10 +2,7 @@
   "name": "geostyler-wfs-parser",
   "version": "0.0.1",
   "description": "GeoStyler Data Parser implementation for Web Feature Service (WFS)",
-  "main": "index.js",
-  "files": [
-    "dist"
-  ],
+  "main": "dist/WfsDataParser.js",
```
Comment (chrismayer) [inline package.json]: Are you sure this is correct? I expect that the build files will be at ``dist/src/WfsDataParser.js``. Did you do a local build to ensure this setting is correct?

File: src/WfsDataParser.ts [hunk 2]
```diff
@@ -1,22 +1,140 @@
...
+
+    const requestParams = Object.keys(params)
+      .map(key => encodeURIComponent(key) + '=' + encodeURIComponent(params[key])).join('&');
+
+    const options = {
+      tagNameProcessors: [this.tagNameProcessor]
+    };
+
+    // Fetch data schema via describe feature type
+    const requestDescribeFeatureTpye = `${url}?${requestParams}`;
```
[the remark this answers is not in the document]
Comment (ahennr) [reply src/WfsDataParser.ts]: See [here](https://github.com/terrestris/geostyler-wfs-parser/pull/3/files#diff-4cf0e4d98d97144cca74220d4e8c338fR113)

File: src/WfsDataParser.ts [hunk 3]
```diff
@@ -1,22 +1,140 @@
...
+          } catch (error) {
+            reject(`Could not parse XML document: ${error}`);
+          }
+        });
     });
 
-    return promise;
+    const fc: FeatureCollection = {
+      type: 'FeatureCollection',
+      features: []
```
[the remark this answers is not in the document]
Comment (ahennr) [reply src/WfsDataParser.ts]: I splitted schema resp. exampleFeature fetching / parsing and in two separate PRs.

Fetching `exampleFeatures` is covered in https://github.com/terrestris/geostyler-wfs-parser/pull/3/

File: package.json (hunk 1 above)
Comment (ahennr) [reply package.json]: Yes, I double-checked this.

In `tsconfig.json` the `dist` folder is configured as output directory and `src` as root directory of the project.

See here
https://github.com/terrestris/geostyler-wfs-parser/blob/36ff9a08c2b4752f139fe9d55f0f764ebf4d2d9f/tsconfig.json#L4

and here

https://github.com/terrestris/geostyler-wfs-parser/blob/36ff9a08c2b4752f139fe9d55f0f764ebf4d2d9f/tsconfig.json#L11

**Comment by chrismayer** [comment]
Perfect, thanks for clarification @ahennr. Feel free to merge.

**Pull request merged**